src/fu-tool.c

```diff
@@ -236,6 +237,15 @@ fu_util_start_engine(FuUtilPrivate *priv, FuEngineLoadFlags flags, GError **erro
 #ifdef HAVE_SYSTEMD
 	g_autoptr(GError) error_local = NULL;
 #endif
+
+#ifdef HAVE_GETUID
+	/* ensure root user */
+	if (priv->interactive && (getuid() != 0 || geteuid() != 0)) {
+		/* TRANSLATORS: we're poking around as a power user */
+		g_printerr("%s\n", _("This program may only work correctly as root"));
```

Comment:
Just a thought - what if we move this to error path instead?

Reply:
I thought about this a bit, but I think we want to warn *before* we throw up warnings on the console.